src/ua_securechannel.c

```diff
@@ -376,9 +376,8 @@ UA_SecureChannel_sendAsymmetricOPNMessage(UA_SecureChannel *channel, UA_UInt32 r
         }
     }
 
-    /* Encrypt message if mode not none */
-    if(channel->securityMode == UA_MESSAGESECURITYMODE_SIGN ||
-       channel->securityMode == UA_MESSAGESECURITYMODE_SIGNANDENCRYPT) {
+    /* Encrypt message if mode is SignAndEncrypt */
+    if(channel->securityMode == UA_MESSAGESECURITYMODE_SIGNANDENCRYPT) {
```

Comment:
Since symmetric keys for signing need to be exchanged while opening a securechannel, encryption has to be present as well, even if the mode is only sign. This change needs to be reverted.

Specification part 7 6.7.4
The OpenSecureChannel Messages are signed and encrypted if the SecurityMode is not None (even if the SecurityMode is SignOnly).

Reply:
Right. Then we can merge the if-condition with the previous one. (This was marked as an issue by PVS Studio.)
I'll include the reference to the spec in a comment.